### PR TITLE
Removed deviceId constraint from getUserMedia()

### PR DESCRIPTION
**Pull request by benmackerras**
Bug fix for firefox OverconstrainedError. Since we were just setting deviceId to 'default' I think its ok to remove.

### CODE REVIEW

**Comment by benmackerras** [comment]
In reference to https://github.com/closeio/mic-recorder-to-mp3/issues/3

**Comment by djalmaaraujo** [comment]
@buzinas 👆

**Comment by buzinas** [comment]
>  Since we were just setting deviceId to 'default' I think its ok to remove.

We are setting `deviceId` to `'default'` as default, but this config exists to be overwritten by the consumer. Check `Object.assign(this.config, config)`. This PR would break the current code.

Btw, I can't reproduce the problem, what are the steps @benmackerras?

File: src/mic-recorder.js [hunk 1]
```diff
@@ -90,10 +90,10 @@ class MicRecorder {
     this.config.sampleRate = this.context.sampleRate;
     this.lameEncoder = new Encoder(this.config);
 
-    const audio = { deviceId: { exact: this.config.deviceId } };
+    const audio = this.config.deviceId ? { deviceId: { exact: this.config.deviceId } } : true;
 
     return new Promise((resolve, reject) => {
-      navigator.mediaDevices.getUserMedia({ audio })
+      navigator.mediaDevices.getUserMedia({ audio: audio })
```
Comment (buzinas) [inline src/mic-recorder.js]: This is unnecessary.

Comment (benmackerras) [reply src/mic-recorder.js]: Not quite sure what you mean sorry, which part is unnecessary?

Comment (buzinas) [reply src/mic-recorder.js]: You're adding `: audio` to this PR, but this is a shorthand notation, thus it's not necessary.

Comment (benmackerras) [reply src/mic-recorder.js]: I found i was getting a Type error when specifying a deviceId and not using `audio: audio`

Comment (buzinas) [reply src/mic-recorder.js]: I mean that `{ audio }` and `{ audio: audio }` are exactly the same.

Comment (benmackerras) [reply src/mic-recorder.js]: I need to specify either `audio: true` or `audio: { deviceId : { exact: this.config.deviceId } }` latest commit makes it a bit more clear

Comment (benmackerras) [reply src/mic-recorder.js]: omg sorry i didnt read the link, i get what you mean now haha

File: src/mic-recorder.js [hunk 2]
```diff
@@ -90,10 +90,10 @@ class MicRecorder {
     this.config.sampleRate = this.context.sampleRate;
     this.lameEncoder = new Encoder(this.config);
 
-    const audio = { deviceId: { exact: this.config.deviceId } };
+    const audio = this.config.deviceId ? { deviceId: { exact: this.config.deviceId } } : true;
 
     return new Promise((resolve, reject) => {
-      navigator.mediaDevices.getUserMedia({ audio })
+      navigator.mediaDevices.getUserMedia({audio})
```
Comment (buzinas) [inline src/mic-recorder.js]:
```suggestion
      navigator.mediaDevices.getUserMedia({ audio })
```

**Pull request merged**